src/components/BlogList/BlogList.jsx

```diff
@@ -1,14 +1,27 @@
 import React from "react";
+import { useSelector } from "react-redux";
 import BlogCard from "@components/BlogCard/BlogCard";
 import "@components/BlogList/blogList.scss";
+import NotFound from "@components/NotFound/NotFound";
 
 const BlogList = ({ blogs }) => {
+  const queryString = useSelector((state) => state?.search?.searchQuery);
+  const searchedBlogs = blogs?.filter((blog) =>
```

Comment:
`blogs` as the name suggests should be a collection of blogs either having some or none blog. It should never be `undefined` or `null`
```suggestion
  const searchedBlogs = blogs.filter((blog) =>
```

Reply:
got it thanks. resolved